src/multiagent-system/modules/common/models/webWorker.ts

```diff
@@ -37,24 +38,14 @@ export default abstract class WebWorker<T extends Unit> {
     }
 
     private setPonderingData(ponderingData: PonderingData): void {
-        this.nearestUnits = [];
-        const constructor = this.getConstructor(ponderingData.unitPackage);
-        this.unit = new constructor(ponderingData.unitPackage.data);
+        this.unit = this.mappers.map(ponderingData.unitPackage);
```

Comment:
Has a problem here. Can't set `Unit` to type `T`, though it extends `Unit`. Need and idea from you 😄

Reply:
In TS it's not allowed to assign a 'concrete type' to a generic Type Parameter.

Reply:
A possible solution in our situation is explicit conversions